plugins/ar_spectral.py

```diff
@@ -803,15 +819,35 @@ def __init__(self, microscope, main_app):
         bigslit = model.getComponent(role="slit-in-big")
         lsw = model.getComponent(role="lens-switch")
 
+        # This is a little tricky: we don't directly need the spectrometer, the
+        # 1D image of the CCD, as we are interested in the raw image. However,
+        # we care about the wavelengths and the spectrometer might be inverted
+        # in order to make sure the wavelength is is the correct direction (ie,
+        # lowest pixel = lowest wavelength). So we need to do the same on the
+        # raw image. However, there is no "official" way to connect the
+        # spectrometer(s) to their raw CCD. So we rely on the fact that
+        # typically this is a wrapper, so we can check using the .dependencies.
+        wl_inverted = False
+        try:
+            spec = self._find_spectrometer(self.ccd)
+        except LookupError as ex:
+            logging.warning("%s, expect that the wavelengths are not inverted", ex)
+        else:
+            # Found spec => check transpose in X (1 or -1), and invert if it's inverted (-1)
+            try:
+                wl_inverted = (spec.transpose[0] == -1)
+            except Exception as ex:
+                logging.warning("%s: expect that the wavelengths are not inverted", ex)

```

Comment:
We need to know if wl is inverted or not. Why would we get an exception? (just for warning the user?)